### PR TITLE
YONK-16: Added split module store for tests

### PR DESCRIPTION
**Pull request by aamishbaloch**

### CODE REVIEW

**Comment by aamishbaloch** [comment]
@ziafazal Can you review this?

File: edx_solutions_api_integration/courses/tests.py [hunk 1]
```diff
@@ -133,44 +138,44 @@ def setUpClass(cls):
         cls.chapter = ItemFactory.create(
             category="chapter",
             parent_location=cls.course.location,
-            data=cls.test_data,
+            # data=cls.test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: Commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 2]
```diff
@@ -133,44 +138,44 @@ def setUpClass(cls):
...
             due=cls.course_end_date,
-            display_name="Overview",
+            display_name="Overview Chapter"
         )
 
         cls.course_project = ItemFactory.create(
             category="chapter",
             parent_location=cls.course.location,
-            data=cls.test_data,
+            # data=cls.test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code.

File: edx_solutions_api_integration/courses/tests.py [hunk 3]
```diff
@@ -133,44 +138,44 @@ def setUpClass(cls):
...
             display_name="Group Project2"
         )
 
         cls.course_content2 = ItemFactory.create(
             category="sequential",
             parent_location=cls.chapter.location,
-            data=cls.test_data,
-            display_name="Sequential",
+            # data=cls.test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 4]
```diff
@@ -133,44 +138,44 @@ def setUpClass(cls):
...
+            display_name="Sequential"
         )
 
         cls.content_child2 = ItemFactory.create(
             category="vertical",
             parent_location=cls.course_content2.location,
-            data=cls.test_data,
+            # data=cls.test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 5]
```diff
@@ -133,44 +138,44 @@ def setUpClass(cls):
...
             display_name="Vertical Sequence"
         )
 
         cls.course_content = ItemFactory.create(
             category="videosequence",
             parent_location=cls.content_child2.location,
-            data=cls.test_data,
-            display_name="Video_Sequence",
+            # data=cls.test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 6]
```diff
@@ -184,14 +189,14 @@ def setUpClass(cls):
             category="video",
             parent_location=cls.content_child2.location,
             data=cls.test_data,
-            display_name="Child Video",
+            display_name="Child Video"
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: why removing trailing comma?

File: edx_solutions_api_integration/courses/tests.py [hunk 7]
```diff
@@ -219,21 +224,22 @@ def setUpClass(cls):
         cls.sub_section = ItemFactory.create(
             parent_location=cls.chapter.location,
             category="sequential",
-            display_name=u"test subsection",
+            # data=cls.test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 8]
```diff
@@ -219,21 +224,22 @@ def setUpClass(cls):
...
+            display_name=u"test subsection"
         )
 
         cls.unit = ItemFactory.create(
             parent_location=cls.sub_section.location,
             category="vertical",
             metadata={'graded': True, 'format': 'Homework'},
-            display_name=u"test unit",
+            display_name=u"test unit"
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: why removing trailing comma.

File: edx_solutions_api_integration/courses/tests.py [hunk 9]
```diff
@@ -302,24 +308,26 @@ def _find_item_by_class(self, items, class_name):
...
+            end=datetime(2015, 1, 16, 14, 30, tzinfo=pytz.UTC)
         )
 
         chapter = ItemFactory.create(
             category="chapter",
             parent_location=course.location,
-            data=self.test_data,
-            due=datetime(2014, 5, 16, 14, 30),
-            display_name="Overview"
+            # data=self.test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 10]
```diff
@@ -302,24 +308,26 @@ def _find_item_by_class(self, items, class_name):
...
+            due=datetime(2014, 5, 16, 14, 30, tzinfo=pytz.UTC),
+            display_name="Overview chapter 2"
         )
 
         sub_section = ItemFactory.create(
             parent_location=chapter.location,
             category="sequential",
+            # data=self.test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 11]
```diff
@@ -355,7 +363,7 @@ def _setup_courses_completions_leaders(self):
             local_content = ItemFactory.create(
                 category="videosequence",
                 parent_location=unit.location,
-                data=self.test_data,
+                # data=self.test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 12]
```diff
@@ -775,7 +784,10 @@ def test_courses_overview_get_unparsed(self):
         response = self.do_get(test_uri)
         self.assertEqual(response.status_code, 200)
         self.assertGreater(len(response.data), 0)
-        self.assertEqual(response.data['overview_html'], self.overview.data)
+
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: what is the reason for additional lines here?

File: edx_solutions_api_integration/courses/tests.py [hunk 13]
```diff
@@ -785,27 +797,25 @@ def test_courses_overview_get_parsed(self):
         self.assertGreater(len(response.data), 0)
         self.assertIn(self.course.course_image, response.data['course_image_url'])
         sections = response.data['sections']
-        self.assertEqual(len(sections), 5)
+
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: additional line?

File: edx_solutions_api_integration/courses/tests.py [hunk 14]
```diff
@@ -785,27 +797,25 @@ def test_courses_overview_get_parsed(self):
...
         self.assertEqual(staff[0]['name'], "Staff Member #1")
-        self.assertEqual(staff[0]['image_src'], "/images/pl-faculty.png")
+        self.assertIn("images_placeholder-faculty.png", staff[0]['image_src'])
         self.assertIn("<p>Biography of instructor/staff member #1</p>", staff[0]['bio'])
         self.assertEqual(staff[1]['class'], "teacher")
         self.assertEqual(staff[1]['name'], "Staff Member #2")
-        self.assertEqual(staff[1]['image_src'], "/images/pl-faculty.png")
+        self.assertIn("images_placeholder-faculty.png", staff[1]['image_src'])
         self.assertIn("<p>Biography of instructor/staff member #2</p>", staff[1]['bio'])
-        self.assertEqual(staff[2]['class'], "author")
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: what is the reason for removing this and next assertion?

File: edx_solutions_api_integration/courses/tests.py [hunk 15]
```diff
@@ -1994,7 +2013,7 @@ def test_coursemodulecompletions_filters(self):  # pylint: disable=R0915
             local_content = ItemFactory.create(
                 category="videosequence",
                 parent_location=self.content_child2.location,
-                data=self.test_data,
+                # data=self.test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code.

File: edx_solutions_api_integration/courses/tests.py [hunk 16]
```diff
@@ -2259,7 +2281,7 @@ def test_course_project_list(self):
...
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 17]
```diff
@@ -2309,7 +2331,7 @@ def test_courses_data_metrics(self):
...
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 18]
```diff
@@ -2662,8 +2685,8 @@ def test_course_navigation(self):
                 'vertical': unicode(self.content_child2.location),
                 'section': unicode(self.course_content2.location),
                 'course_key': unicode(self.course.id),
-                'final_target_id': unicode(self.content_subchild.location),
-                'position': '1',
+                'final_target_id': unicode(self.content_child.location),
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: reason for changing `content_subchild` to `content_child`

File: edx_solutions_api_integration/courses/tests.py [hunk 19]
```diff
@@ -2743,13 +2767,14 @@ def setUpClass(cls):
         cls.chapter = ItemFactory.create(
             category="chapter",
             parent_location=cls.course.location,
-            data=cls.test_data,
+            # data=cls.test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 20]
```diff
@@ -2743,13 +2767,14 @@ def setUpClass(cls):
...
             due=course_end_date,
             display_name=u"3033 Overview"
         )
         cls.sub_section = ItemFactory.create(
             parent_location=cls.chapter.location,
             category="sequential",
+            # data=cls.test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 21]
```diff
@@ -2761,15 +2786,15 @@ def setUpClass(cls):
         cls.item = ItemFactory.create(
             parent_location=cls.unit.location,
             category='problem',
-            data=StringResponseXMLFactory().build_xml(answer='bar'),
+            # data=StringResponseXMLFactory().build_xml(answer='bar'),
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py [hunk 22]
```diff
@@ -2761,15 +2786,15 @@ def setUpClass(cls):
...
             display_name='Problem to test timeseries',
             metadata={'rerandomize': 'always', 'graded': True, 'format': 'Midterm Exam'}
         )
 
         cls.item2 = ItemFactory.create(
             parent_location=cls.unit.location,
             category='problem',
-            data=StringResponseXMLFactory().build_xml(answer='bar'),
+            # data=StringResponseXMLFactory().build_xml(answer='bar'),
```
Comment (ziafazal) [inline edx_solutions_api_integration/courses/tests.py]: commented code

File: edx_solutions_api_integration/groups/tests.py [hunk 23]
```diff
@@ -51,7 +53,7 @@ def setUp(self):
         self.course_content = ItemFactory.create(
             category="videosequence",
             parent_location=self.course.location,
-            data=self.test_course_data,
+            # data=self.test_course_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/groups/tests.py]: commented code

File: edx_solutions_api_integration/test_utils.py [hunk 24]
```diff
@@ -93,60 +93,60 @@ def setup_course_with_grading(self, start=None, end=None):
         chapter1 = ItemFactory.create(
             category="chapter",
             parent_location=grading_course.location,
-            data=test_data,
+            # data=test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/test_utils.py]: commented code

File: edx_solutions_api_integration/test_utils.py [hunk 25]
```diff
@@ -93,60 +93,60 @@ def setup_course_with_grading(self, start=None, end=None):
...
             display_name="Chapter 1"
         )
         chapter2 = ItemFactory.create(
             category="chapter",
             parent_location=grading_course.location,
-            data=test_data,
+            # data=test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/test_utils.py]: commented code

File: edx_solutions_api_integration/test_utils.py [hunk 26]
```diff
@@ -93,60 +93,60 @@ def setup_course_with_grading(self, start=None, end=None):
...
             display_name="Sequence 2",
             graded=True,
             metadata={'rerandomize': 'always', 'graded': True, 'format': "Homework"}
         )
         vertical1 = ItemFactory.create(
             category="vertical",
             parent_location=sequential2.location,
-            data=test_data,
+            # data=test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/test_utils.py]: commented code

File: edx_solutions_api_integration/test_utils.py [hunk 27]
```diff
@@ -93,60 +93,60 @@ def setup_course_with_grading(self, start=None, end=None):
...
             display_name="Vertical 1",
         )
         sequential3 = ItemFactory.create(
             category="sequential",
             parent_location=chapter2.location,
-            data=test_data,
+            # data=test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/test_utils.py]: commented code

File: edx_solutions_api_integration/test_utils.py [hunk 28]
```diff
@@ -93,60 +93,60 @@ def setup_course_with_grading(self, start=None, end=None):
...
             display_name="Sequence 3",
             graded=True,
             metadata={'rerandomize': 'always', 'graded': True, 'format': "Midterm Exam"}
         )
         vertical2 = ItemFactory.create(
             category="vertical",
             parent_location=sequential3.location,
-            data=test_data,
+            # data=test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/test_utils.py]: commented code

File: edx_solutions_api_integration/test_utils.py [hunk 29]
```diff
@@ -93,60 +93,60 @@ def setup_course_with_grading(self, start=None, end=None):
...
             display_name="Vertical 2",
         )
         item = ItemFactory.create(
             parent_location=vertical1.location,
             category='mentoring',
-            data=StringResponseXMLFactory().build_xml(answer='foo'),
+            # data=StringResponseXMLFactory().build_xml(answer='foo'),
```
Comment (ziafazal) [inline edx_solutions_api_integration/test_utils.py]: commented code

File: edx_solutions_api_integration/test_utils.py [hunk 30]
```diff
@@ -93,60 +93,60 @@ def setup_course_with_grading(self, start=None, end=None):
...
             display_name=u"test mentoring homework",
 
         )
         item2 = ItemFactory.create(
             parent_location=vertical2.location,
             category='mentoring',
-            data=StringResponseXMLFactory().build_xml(answer='bar'),
+            # data=StringResponseXMLFactory().build_xml(answer='bar'),
```
Comment (ziafazal) [inline edx_solutions_api_integration/test_utils.py]: commented code

File: edx_solutions_api_integration/users/tests.py [hunk 31]
```diff
@@ -1259,49 +1276,49 @@ def test_user_courses_detail_post_position_course_as_descriptor(self):
         chapter1 = ItemFactory.create(  # pylint: disable=W0612
             category="chapter",
             parent_location=course.location,
-            data=test_data,
+            # data=test_data,
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/tests.py]: commented code

File: edx_solutions_api_integration/courses/tests.py (hunk 14 above)
Comment (aamishbaloch) [reply edx_solutions_api_integration/courses/tests.py]: We are now getting only two staff members, and both are teachers.

File: edx_solutions_api_integration/courses/tests.py (hunk 18 above)
Comment (aamishbaloch) [reply edx_solutions_api_integration/courses/tests.py]: Getting content_child as final_target_id now.

**Comment by aamishbaloch** [comment]
@ziafazal Update with the suggested changes. Can you review the changes?

**Pull request merged**